backend/api/request_create.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"database/sql"
+	"errors"
+	"net/http"
+
+	db "github.com/git-adithyanair/cs130-group-project/db/sqlc"
+	"github.com/git-adithyanair/cs130-group-project/token"
+
+	"github.com/gin-gonic/gin"
+)
+
+type CreateRequestRequestItem struct {
+	Name           string              `json:"name" binding:"required,min=1"`
+	QuantityType   db.ItemQuantityType `json:"quantity_type" binding:"required,min=2"`
+	Quantity       float64             `json:"quantity" binding:"required,min=1"`
+	PreferredBrand string              `json:"preferred_brand"`
+	Image          string              `json:"image"`
+	ExtraNotes     string              `json:"extra_notes"`
+}
+
+type CreateRequestRequest struct {
+	CommunityID int64                      `json:"community_id" binding:"required,min=1"`
+	StoreID     *int64                     `json:"store_id"`
+	Items       []CreateRequestRequestItem `json:"items" binding:"required"`
+}
+
+func (server *Server) CreateRequest(ctx *gin.Context) {
+	var req CreateRequestRequest
+	if err := ctx.ShouldBindJSON(&req); err != nil {
+		ctx.JSON(http.StatusBadRequest, errorResponse(err))
+		return
+	}
+
+	authPayload := ctx.MustGet(authorizationPayloadKey).(*token.Payload)
+
+	if _, err := server.queries.GetCommunity(ctx, req.CommunityID); err != nil {
+		if err == sql.ErrNoRows {
+			err := errors.New("No community exists with given ID.")
+			ctx.JSON(http.StatusNotFound, errorResponse(err))
+		} else {
+			ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+		}
+		return
+	}
+
+	if len(req.Items) == 0 {
+		err := errors.New("Request must contain at least one item.")
+		ctx.JSON(http.StatusNotFound, errorResponse(err))
+		return
+	}
+
+	arg := db.CreateRequestParams{
+		UserID:      authPayload.UserID,
+		CommunityID: sql.NullInt64{Int64: req.CommunityID, Valid: true},
+	}
+	if req.StoreID != nil {
+		if _, err := server.queries.GetStore(ctx, *req.StoreID); err != nil {
+			if err == sql.ErrNoRows {
+				err := errors.New("No store exists with given ID.")
+				ctx.JSON(http.StatusNotFound, errorResponse(err))
+			} else {
+				ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+			}
+			return
+		}
+
+		arg = db.CreateRequestParams{
```

Comment:
Is it possible to just do `arg.StoreID = sql.NullInt64{Int64: *req.StoreID, Valid: true}` here?

Reply:
oh yes did not know you could do that but it seems to work